packages/helpers/src/modules/assetValue.ts

```diff
@@ -292,28 +312,45 @@ function getAssetString(assetOrChain: AssetIdentifier) {
 
 function getAssetInfo(identifier: string) {
   const shortIdentifier = identifier.slice(0, 14);
-  const splitIdentifier = identifier.split(".");
-  const identifierChain = splitIdentifier[0]?.toUpperCase() as Chain;
-  const isThorOrMaya = [Chain.THORChain, Chain.Maya].includes(identifierChain);
-
   const isSynthetic = shortIdentifier.includes("/");
   const isTradeAsset = shortIdentifier.includes("~");
   const isSynthOrTrade = isSynthetic || isTradeAsset;
-  const assetSeparator = isTradeAsset ? "~" : "/";
 
-  const [synthChain, synthSymbol = ""] = isThorOrMaya
-    ? splitIdentifier.slice(1).join().split(assetSeparator)
-    : identifier.split(assetSeparator);
+  // Logic for Synthetic and Trade Assets
+  if (isSynthOrTrade) {
+    const splitIdentifier = identifier.split(".");
+    const identifierChain = splitIdentifier[0]?.toUpperCase() as Chain;
+    const isThorOrMaya = [Chain.THORChain, Chain.Maya].includes(identifierChain);
+    const assetSeparator = isTradeAsset ? "~" : "/";
 
-  if (isSynthOrTrade && !(synthChain && synthSymbol)) {
-    throw new SwapKitError({ errorKey: "helpers_invalid_asset_identifier", info: { identifier } });
-  }
+    const [synthChain, synthSymbol = ""] = isThorOrMaya
+      ? splitIdentifier.slice(1).join(".").split(assetSeparator)
+      : identifier.split(assetSeparator);
 
-  const [chain, ...rest] = (
-    identifier.includes(".") && !isSynthOrTrade ? identifier : `${identifierChain}.${synthSymbol}`
-  ).split(".") as [Chain, string];
+    if (!(synthChain && synthSymbol)) {
+      throw new SwapKitError({ errorKey: "helpers_invalid_asset_identifier", info: { identifier } });
+    }
 
-  const assetSymbol = isSynthOrTrade ? synthSymbol : rest.join(".");
+    // Get the ticker from the base symbol (e.g., "AVAX" from "AVAX/AVAX")
+    const { ticker, address } = getAssetBaseInfo({ chain: synthChain as Chain, symbol: synthSymbol });
+    const finalSymbol = `${synthChain}${assetSeparator}${synthSymbol}`;
+
+    return {
+      address,
+      chain: identifierChain,
+      isGasAsset: false,
+      isSynthetic,
+      isSynthOrTrade,
+      isTradeAsset,
+      symbol: finalSymbol,
+      ticker, // This is now correct (e.g., "AVAX")
+    };
+  }
```

Comment:
Adding additional branching and handling "different" returns for synths and normal assets already sparks that those should be 2 different functions - can share "common" parameter parsing but I would extract both this and after this if statements.

I'm also not a fan of "early" returns unless they are one-liners, but that's left for you to decide :)